test/functional/fixtures/regression/gh-6722/test.js

```diff
@@ -0,0 +1,75 @@
+const expect                                                    = require('chai').expect;
+const { SUCCESS_RESULT_ATTEMPTS, FAIL_RESULT_ATTEMPTS, ERRORS } = require('./constants');
+const { createReporter }                                        = require('../../../utils/reporter');
+
+const getCustomReporter = function (result) {
+    return createReporter({
+        reportTestDone: function (name, { errs, quarantine, browsers }) {
+            if (quarantine)
+                Object.assign(result, { quarantine, browsers });
+
+            if (errs && errs.length > 0)
+                throw new Error(errs[0]['message']);
+        },
+    });
+};
+
+const expectAttempts = (attempts, { quarantine, browsers }) => {
+    const attemptsCount        = attempts.length;
+    const successAttemptsCount = attempts.filter(attempt => attempt === ERRORS.None).length;
+    const failedAttemptsCount  = attempts.filter(attempt => attempt !== ERRORS.None).length;
+    const browsersCount        = browsers.length;
+
+    const currentAttemptsCount        = Object.keys(quarantine).length;
+    const currentSuccessAttemptsCount = Object.values(quarantine).filter(attempt => attempt.passed).length;
+    const currentFailedAttemptsCount  = Object.values(quarantine).filter(attempt => !attempt.passed).length;
+    const currentTestRunIdsCount      = browsers.reduce((counter, browser) => counter + browser.quarantineAttemptsTestRunIds.length, 0);
+
+    // We have to add the number of attempts per browser as we also kept the old behavior. {1:{}, 2:{},  ... testRunId1:{}, testRunId2:{}, ...}
+    const expectedAttemptsCount        = attemptsCount * (browsersCount + 1);
+    const expectedSuccessAttemptsCount = successAttemptsCount * (browsersCount + 1);
+    const expectedFailedAttemptsCount  = failedAttemptsCount * (browsersCount + 1);
+
+    return expect(currentAttemptsCount).to.equal(expectedAttemptsCount) &&
```

Comment:
No need to return anything from this function.
No need to add && between conditions.
It's better to write assertions line by line:
```
        expect(currentSuccessAttemptsCount).to.equal(expectedSuccessAttemptsCount);
        expect(currentFailedAttemptsCount).to.equal(expectedFailedAttemptsCount);
        expect(currentTestRunIdsCount).to.equal(attemptsCount * browsersCount);
```